src/Akeneo/Category/front/src/feature/components/channel/ChannelSelector.tsx

```diff
@@ -38,7 +38,7 @@ const ChannelSelector = ({value, values, onChange}: ChannelSelectorProps) => {
   return (
     <Dropdown>
       <SwitcherButton inline onClick={() => {open();}} label={translate('pim_common.channel')}>
-        <ChannelSpan code={`[${selectedChannel.code}]`} label={selectedChannel.labels[catalogLocale]} />
+        <ChannelSpan code={`[${selectedChannel?.code}]`} label={selectedChannel?.labels[catalogLocale]} />
```

Comment:
A more resilient way to handle the case there is no selected value could be
```suggestion
        <span>{selectedChannel && (selectedChannel.label ?? `[${selectedChannel.code}]`}</span>
```

Furthermore, you can use the helper [getLabel](https://github.com/akeneo/pim-community-dev/blob/master/front-packages/shared/src/models/label-collection.ts) in the shared library to display the label

Reply:
Thx for the tip, it has been improved